prow/scripts/resources/README.md

```diff
@@ -8,11 +8,53 @@ The folder contains files that are directly used by Prow pipeline scripts.
 ├── kube-dns-stub-domains-patch.yaml                          # Enables the build.kyma-project.io stubdomain and provides Google root DNS servers IPs.
 ├── limitrange-patch.yaml                                     # Increases the kyma-system Namespace maximum memory request for containers.
 ├── prometheus-operator-stackdriver-patch.yaml                # Injects the Stackdriver collector sidecar, sets metric filters, and enables scraping Stackdriver target.
-└── prometheus-operator-additional-scrape-config.yaml         # Additional scrape configuration for Prometheus operator.
+├── prometheus-operator-additional-scrape-config.yaml         # Additional scrape configuration for Prometheus operator.
+└── debug-container.yaml                                      # Create pod and configmap for collecting memory related statistics for k8s node and containers.
 ```
 
 ## Prometheus operator additional scrape configuration
 
 Prometheus operator expects to have additional scrape configuration provided as a Secret. This Secret is appended to the Prometheus scrape config file.
 Additional scrape config allows you to add scrape targets outside automatic scrape targets discovery mechanisms.
 It is administrator's responsibility to provide syntactically correct scrape configuration.
+
+## Troubleshoot OOM events with debug-container.
+
+Standard OOM events logged in os logs and k8s node events provide details about allocated memory, PID and process name which was killed to release memory. Killed process, due to OOM activity not always is the same which consume too much memory. Even if killed process is the one which consumed too much memory, it's hard to match its name with pods running on k8s node. For that purpose use a debug-container.yaml. It contains configmap with bash script and pod where this script is running. A pod is able to access host `/proc` and `/sys` filesystems and docker daemon socket. Once scheduled, it will collect host and running containers memory stats. Apart memory stats, for all containers it will collect container PID, container name and all PIDs belong to the same cgroup. Collected data should be copied to the artifacts directory on GCP bucket with prowjob logs.
```

Comment:
```suggestion
Standard OOM events logged in the OS logs and k8s node events provide details about the allocated memory, PID, and the process name which was killed to release memory. Handling OOM events may result in killing a process different than the one which consumed too much memory. Even if the killed process is the one which consumed too much memory, it's difficult to match its name with pods running on k8s Node. For that purpose, use the `debug-container.yaml` file. It contains a ConfigMap with bash script and a pod where this script is running. A pod is able to access the host's `/proc` and `/sys` filesystems and the docker daemon socket. Once scheduled, it will collect the host, running containers memory stats, container PID, container name for all containers, and all PIDs that belong to the same cgroup. Collected data should be copied to the `artifacts` directory on GCP bucket with ProwJob logs.
```